test/ui/TextInputTest.ts

```diff
@@ -40,6 +44,31 @@ export function TextInputTest() {
         textInput.setValue(value);
         expect(textInput.getValue()).toEqual(value);
       });
+
+      it('should call the onchange', () => {
+        textInput.setValue('test');
+        expect(spy).toHaveBeenCalledWith(textInput);
+      });
+
+      it('should not call the onchange multiple time', () => {
```

Comment:
Missing tests for 

* should call onChange twice with two `setValue`
* should call onChange with `setValue` and empty value

?